### PR TITLE
Restructure API docs - Closes #1438

### LINKED ISSUE #1438: Restructure API docs
### Description

The Description of Lisk nodes RPC endpoints should be moved to the `Understand blockchain` category.

Description of application actions & events is moved to general docs under API category

### PR DESCRIPTION
**Pull request by Tschakki**
Closes #1438

### CODE REVIEW

File: docs/modules/ROOT/pages/api/lisk-node-rpc.adoc [hunk 1]
```diff
@@ -1,305 +1,1742 @@
...
+:url_general_endpoints: understand-blockchain/endpoints.adoc
+:url_modules: {docs_sdk}modules/index.adoc
+:url_plugins: {docs_sdk}plugins/index.adoc
 
-The open Lisk communication architecture is based on two different RPC (Remote-Procedure-Call), API modes: {url_wiki_ipc}[Inter-Process Communication (IPC)^] and {url_websocket}[WebSocket (WS)^].
-The xref:{url_advanced_architecture}[Application] can be configured to either expose an IPC or a WS API that can be used by internal components such as modules and plugins, as well as by any external service such as other scripts in JS, a tool in Rust, or a Python daemon.
+A reference of all actions and events that are provided by the Lisk Framework for a blockchain application.
 
-TIP: For more information about the configuration of the RPC endpoints, check out the xref:{url_guides_config_rpc}[configuration guide].
+TIP: For mor information about the public application endpoints, check out the xref:{url_general_endpoints}[] page.
```
Comment (chris529) [inline docs/modules/ROOT/pages/api/lisk-node-rpc.adoc]:
```suggestion
TIP: For more information about the public application endpoints, check out the xref:{url_general_endpoints}[] page.
```

File: docs/modules/ROOT/pages/understand-blockchain/endpoints.adoc [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+		process.exit(0);
+	});
+});
+----
+--
+=====
+
+=== IPC vs WS
+
+There are two ways through which a node can communicate via the API:
```
Comment (chris529) [inline docs/modules/ROOT/pages/understand-blockchain/endpoints.adoc]:
```suggestion
There are two methods available whereby a node can communicate via the API:
```

File: docs/modules/ROOT/pages/understand-blockchain/endpoints.adoc [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+
+* IPC (Inter-Process-Communication)
+* WS (WebSocket)
+
+In general, IPC is preferred for local connections because it...
```
Comment (chris529) [inline docs/modules/ROOT/pages/understand-blockchain/endpoints.adoc]:
```suggestion
In general, IPC is the preferred method for local connections for the following reasons:
```

File: docs/modules/ROOT/pages/understand-blockchain/endpoints.adoc [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+
+* is slightly faster
```
Comment (chris529) [inline docs/modules/ROOT/pages/understand-blockchain/endpoints.adoc]:
```suggestion
* It is slightly faster
```

File: docs/modules/ROOT/pages/understand-blockchain/endpoints.adoc [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+* supports synchronous data exchange
```
Comment (chris529) [inline docs/modules/ROOT/pages/understand-blockchain/endpoints.adoc]:
```suggestion
* It supports synchronous data exchange
```

File: docs/modules/ROOT/pages/understand-blockchain/endpoints.adoc [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+* does not use the ports of the system, avoiding any risk of collision when the port is already used by another application
```
Comment (chris529) [inline docs/modules/ROOT/pages/understand-blockchain/endpoints.adoc]:
```suggestion
* It does not use the system ports, therefore avoiding any risk of collision when the ports are already in use by another application
```

File: docs/modules/ROOT/pages/understand-blockchain/endpoints.adoc [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+----
+const data = await client.invoke('app:getSchema'); // <1>
+const data = await client.invoke('app:actionName', input); // <2>
+client.invoke('monitor:getTransactionStats').then((val) => { // <3>
+    console.log(val);
+});
+----
+
+<1> How to invoke an action.
+<2> How to invoke an action that needs some input data.
```
Comment (chris529) [inline docs/modules/ROOT/pages/understand-blockchain/endpoints.adoc]:
```suggestion
<2> How to invoke an action that needs some data input.
```

**Review by chris529 — COMMENTED** [review]
Looks good, just a few minor points 👍

**Review by shuse2 — APPROVED** [review]

**Pull request merged**